### ISSUE TITLE
IMAP auth fails for account with dot in name and encrypted password

### ISSUE THREAD

**Issue by rhorns** [thunderbird/thunderbird-android#2039]
### Expected behavior
K-9 to authenticate and pull down email from IMAP server.

### Actual behavior
Authentication fails. ADB log:
```
01-10 20:42:35.579 E/k9      (12589): com.fsck.k9.mail.AuthenticationFailedException: Command: AUTHENTICATE CRAM-MD5; response: #1# [NO, [AUTHENTICATIONFAILED], Authentication failed.]
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapConnection.authCramMD5(ImapConnection.java:379)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapConnection.authenticate(ImapConnection.java:327)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapConnection.open(ImapConnection.java:121)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapConnection.sendCommand(ImapConnection.java:640)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapConnection.executeSimpleCommand(ImapConnection.java:623)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapConnection.executeSimpleCommand(ImapConnection.java:612)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapFolder.executeSimpleCommand(ImapFolder.java:109)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapFolder.internalOpen(ImapFolder.java:145)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapFolderPusher$PushRunnable.openConnectionIfNecessary(ImapFolderPusher.java:330)
01-10 20:42:35.579 E/k9      (12589): 	at com.fsck.k9.mail.store.imap.ImapFolderPusher$PushRunnable.run(ImapFolderPusher.java:155)
01-10 20:42:35.579 E/k9      (12589): 	at java.lang.Thread.run(Thread.java:761)
```

### Steps to reproduce
1. Set up account
Incoming server:
Security: SSL/TLS
Port: 993
Username: name.withdot
Authentication: Encrypted password

### Environment
K-9 Mail version: v5.202

Android version: 7.0

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

Additional info:

1. Another account with identical settings but no dot in the account name at same server works fine.

2. If I turn off password encryption the authentication succeeds.

**Comment by rhorns** [comment]
BTW: This started happening with one of the most recent updates.

**Comment by rhorns** [comment]
This seems to be fixed in the latest version (5.203). Thanks!
